client/src/Widgets/ActivityTable.tsx

```diff
@@ -20,15 +20,210 @@ import {
   MenuButton,
   MenuList,
   Show,
+  LinkBox,
+  LinkOverlay,
 } from "@chakra-ui/react";
 import { GoKebabVertical } from "react-icons/go";
 import { FaFolder } from "react-icons/fa";
 import { RiDraftFill } from "react-icons/ri";
 import { MdAssignment } from "react-icons/md";
 import { BsPeopleFill } from "react-icons/bs";
-import { useFetcher, useNavigate } from "react-router-dom";
+import { useFetcher } from "react-router-dom";
 import { AssignmentStatus } from "../_utils/types";
 
+// this component is separate so that it can have its own states
+function ActivityRow({
+  activity,
+  showAssignmentStatus,
+  showPublicStatus,
+  suppressAvatar,
+  showOwnerName,
+  ref,
+}) {
+  console.log(showPublicStatus);
+
+  const [rowTitle, setRowTitle] = useState(activity.title);
+  const fetcher = useFetcher();
+
+  // from ActivityEditor.tsx
+  let lastActivityDataName = useRef(activity.title);
+  //Update when something else updates the name
+  if (
+    activity.title != lastActivityDataName.current &&
+    rowTitle != activity.title
+  ) {
+    setRowTitle(activity.title);
+  }
+  lastActivityDataName.current = activity.title;
+
+  let assignmentStatusString: string =
+    activity.assignmentStatus !== null &&
+    activity.assignmentStatus !== undefined &&
+    activity.assignmentStatus !== "Unassigned"
+      ? activity.assignmentStatus
+      : "";
+
+  if (
+    activity.assignmentStatus === "Open" &&
+    activity.closeTime !== undefined
+  ) {
+    assignmentStatusString =
+      assignmentStatusString + " until " + activity.closeTime;
+  }
+
+  let tooltipLabelString: string = activity.isFolder
+    ? activity.isPublic || activity.isShared
+      ? "Folder / Public"
+      : "Folder / Private"
+    : activity.authorRow
+      ? activity.ownerName
+      : "Activity" +
+        (activity.assignmentStatus ? " / " + activity.assignmentStatus : "");
+
+  function saveUpdatedTitle() {
+    if (rowTitle !== activity.title && activity.id !== undefined) {
+      fetcher.submit(
+        {
+          _action: "update title",
+          id: activity.id,
+          cardTitle: rowTitle,
+          isFolder: Boolean(activity.isFolder),
+        },
+        { method: "post" },
+      );
+    }
+  }
+
+  return (
+    <LinkBox
+      as={Tr}
+      key={(activity.authorRow ? "author" : "activity") + activity.id}
+      data-test="Activity Link"
+      cursor="pointer"
+      _hover={{ backgroundColor: "#eeeeee" }}
+      borderBottom="2px solid gray"
+      width="100%"
+    >
+      <Td p="0" m="0" width="20px">
+        <LinkOverlay href={activity.cardLink ? activity.cardLink : ""}>
```

Comment:
I really like how this `<LinkBox>`/`<LinkOverlay>` ended up working out!

We do want to use React Router link so that it takes advantage of the single-page app facilities when following links. To use it for `<LinkOverlay>`, you can change line 31 to
```
import { Link as ReactRouterLink, useFetcher } from "react-router-dom";
```
add `as={ReactRouterLink}` to `<LinkOverlay>` and then change the `href={...}` attribute to `to={...}`.